rcl/src/rcl/security.c

```diff
@@ -0,0 +1,356 @@
+    }
+
+    if (!use_security) {
+      security_options->enforce_security = RMW_SECURITY_ENFORCEMENT_PERMISSIVE;
+    } else {  // if use_security
+      // File discovery magic here
+      char * secure_root = rcl_get_secure_root(
+        name,
+        namespace_,
+        allocator);
+      if (secure_root) {
+        RCUTILS_LOG_INFO_NAMED(ROS_PACKAGE_NAME, "Found security directory: %s", secure_root);
+        security_options->security_root_path = secure_root;
+      } else {
+        if (RMW_SECURITY_ENFORCEMENT_ENFORCE == security_options->enforce_security) {
+          return RCL_RET_ERROR;
+        }
+      }
+    }
+  }
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_security_enabled(bool * use_security)
+{
+  const char * ros_security_enable = NULL;
+  const char * get_env_error_str = NULL;
+
+  if (!use_security) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  get_env_error_str = rcutils_get_env(ROS_SECURITY_ENABLE_VAR_NAME, &ros_security_enable);
+  if (NULL != get_env_error_str) {
+    RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "Error getting env var '" RCUTILS_STRINGIFY(ROS_SECURITY_ENABLE_VAR_NAME) "': %s\n",
+      get_env_error_str);
+    return RCL_RET_ERROR;
+  }
+
+  *use_security = (0 == strcmp(ros_security_enable, "true"));
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_get_enforcement_policy(rmw_security_enforcement_policy_t * policy)
+{
+  const char * ros_enforce_security = NULL;
+  const char * get_env_error_str = NULL;
+
+  if (!policy) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  get_env_error_str = rcutils_get_env(ROS_SECURITY_STRATEGY_VAR_NAME, &ros_enforce_security);
+  if (NULL != get_env_error_str) {
+    RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "Error getting env var '" RCUTILS_STRINGIFY(ROS_SECURITY_STRATEGY_VAR_NAME) "': %s\n",
+      get_env_error_str);
+    return RCL_RET_ERROR;
+  }
+
+  *policy = (0 == strcmp(ros_enforce_security, "Enforce")) ?
+    RMW_SECURITY_ENFORCEMENT_ENFORCE : RMW_SECURITY_ENFORCEMENT_PERMISSIVE;
+  return RCL_RET_OK;
+}
+
+/**
+ * A security lookup function takes in the node/context's name, namespace, a security root directory and an allocator;
+ *  It returns the relevant information required to load the security credentials,
+ *   which is currently a path to a directory on the filesystem containing DDS Security permission files.
+ */
+typedef char * (* security_lookup_fn_t) (
+  const char * name,
+  const char * namespace_,
+  const char * ros_secure_root_env,
+  const rcl_allocator_t * allocator
+);
+
+char * exact_match_lookup(
+  const char * name,
+  const char * namespace_,
+  const char * ros_secure_root_env,
+  const rcl_allocator_t * allocator
+);
+
+char * prefix_match_lookup(
+  const char * name,
+  const char * namespace_,
+  const char * ros_secure_root_env,
+  const rcl_allocator_t * allocator
+);
+
+const security_lookup_fn_t g_security_lookup_fns[] = {
+  NULL,
+  exact_match_lookup,
+  prefix_match_lookup,
+};
+
+typedef enum ros_security_lookup_type_e
```

Comment:
Is there a reason to use the suffix `_e` instead of `_t` like the other enums in `rcl`?

Reply:
This comes from the previous `security_directory.h`. I can fix it if you want, but it's unrelated.

P.S.: I don't know why github just shows a deleted file and a new file in the diff.
  If you check the diff of each commit, you would be able to understand better the history (as there is one commit just renaming files). For the same reason, I will avoid squash and merge, and left a separated commit renaming the files.

Reply:
I missed that the file was renamed. We can leave it as-is since the change is unrelated.